### PR TITLE
📛 NFT minting rework

### PR DESCRIPTION
**Pull request by WRadoslaw**
#4102

### CODE REVIEW

File: packages/atlas/src/components/_overlays/MintNftFirstTimeModal/MintNftFirstTimeModal.tsx [hunk 1]
```diff
@@ -2,36 +2,43 @@ import { FC } from 'react'
...
         }}
+        additionalActionsNode={
+          <Checkbox label="Don't show this again" onChange={onShouldHideNextTime} value={shouldHideNextTime} />
+        }
         actionDivider
       >
         <Text as="p" variant="t200" color="colorText">
-          Now that you've minted your first NFT, you can put it on sale by choosing "Start sale" from your video
-          options.
+          You can mint your videos and sell them on the market for JOY. To do that select mint NFT option in context
```
Comment (attemka) [inline packages/atlas/src/components/_overlays/MintNftFirstTimeModal/MintNftFirstTimeModal.tsx]:
```suggestion
          You can mint your videos and sell them on the market for ${atlasConfig.joystream.tokenTicker}. To do that select mint NFT option in context
```

File: packages/atlas/src/components/_overlays/MintNftModal/MintNftModal.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  const handleVideoWorkspaceSubmit = useHandleVideoWorkspaceSubmit()
+  const { setNftToMint, nftToMint, openNftPutOnSale } = useNftActions()
+  const [showSaleConfirmationDialog, setShowSaleConfirmationDialog] = useState(false)
+  const saleConfirmationDismissed = usePersonalDataStore((state) =>
+    state.dismissedMessages.some((message) => message.id === SALE_CONFIRMATION_ID)
+  )
+  const updateMintConfirmationDismiss = usePersonalDataStore((state) => state.actions.updateDismissedMessages)
+
+  const form = useForm<VideoWorkspaceVideoFormFields>({
+    defaultValues: { ...tabData, mintNft: true } ?? undefined,
```
Comment (attemka) [inline packages/atlas/src/components/_overlays/MintNftModal/MintNftModal.tsx]: can't see the option when this object will have falsy value

File: packages/atlas/src/hooks/useVideoForm.ts [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+      return {
+        ...(isNew || dirtyFields.title ? { title: data.title } : {}),
+        ...(isNew || dirtyFields.description ? { description: data.description } : {}),
+        ...(isNew || dirtyFields.category ? { category: data.category } : {}),
+        ...(isNew || dirtyFields.isPublic ? { isPublic: data.isPublic } : {}),
+        ...((isNew || dirtyFields.hasMarketing) && data.hasMarketing != null
+          ? { hasMarketing: data.hasMarketing }
+          : {}),
+        ...((isNew || dirtyFields.isExplicit) && data.isExplicit != null ? { isExplicit: data.isExplicit } : {}),
```
Comment (attemka) [inline packages/atlas/src/hooks/useVideoForm.ts]: packages/atlas/src/providers/videoWorkspace/types.ts:46
 ` isExplicit: boolean`
so it wont be null

File: packages/atlas/src/hooks/useVideoForm.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+        ...((isNew || dirtyFields.language) && data.language != null ? { language: data.language } : {}),
```
Comment (attemka) [inline packages/atlas/src/hooks/useVideoForm.ts]: > data.language != null

why not "!==" ? why would we need a type conversion here if language: string | null?

File: packages/atlas/src/hooks/useVideoForm.ts [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+        ...(isNew || anyLicenseFieldsDirty ? { license } : {}),
+        ...((isNew || dirtyFields.enableComments) && data.enableComments != null
+          ? { enableComments: data.enableComments }
+          : {}),
+        ...((isNew || dirtyFields.publishedBeforeJoystream) &&
+        data.publishedBeforeJoystream != null &&
```
Comment (attemka) [inline packages/atlas/src/hooks/useVideoForm.ts]: same as above

File: packages/atlas/src/hooks/useVideoForm.ts [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+          nftMetadata,
+          assets,
+          [], // provide empty removedAssetsId array to simplify fee calculation
+          dataObjectStateBloatBondValue.toString(),
+          channelBucketsCount.toString(),
+        ]
+      : undefined,
+    assets
+  )
+  // console.log('fee', updateVideoFee, updateVideoFeeLoading, isSigned && isEdit && id)
```
Comment (attemka) [inline packages/atlas/src/hooks/useVideoForm.ts]: log

File: packages/atlas/src/views/studio/MyVideosView/MyVideosView.tsx [hunk 7]
```diff
@@ -324,113 +327,117 @@ export const MyVideosView = () => {
...
+        {!smMatch && sortVisibleAndUploadButtonVisible && (
+          <MobileButton size="large" icon={<SvgActionAddVideo />} fullWidth {...uploadVideoButtonProps}>
+            Upload video
+          </MobileButton>
+        )}
+        {hasNoVideos ? (
+          <EmptyFallback
+            verticalCentered
+            title="Add your first video"
+            subtitle="No videos uploaded yet. Start publishing by adding your first video to Joystream."
```
Comment (attemka) [inline packages/atlas/src/views/studio/MyVideosView/MyVideosView.tsx:347]: @dmtrjsg should it be "Joystream" or the app name here?

**Review by attemka — CHANGES_REQUESTED** [review]

**Review by thesan — COMMENTED** [review]
It took me a while to get around this PR but it does demonstrate a lot of "the Atlas way". I'll probably go back to it later for reference. However I don't have enough context to make any valuable observation.

File: packages/atlas/src/components/_overlays/MintNftModal/MintNftModal.tsx (hunk 2 above)
Comment (WRadoslaw) [reply packages/atlas/src/components/_overlays/MintNftModal/MintNftModal.tsx]: I don't understand the question/assumption. Do you mean that `tabData` will always have `mintNft` on true?

Comment (attemka) [reply packages/atlas/src/components/_overlays/MintNftModal/MintNftModal.tsx]: I mean, it wont go at null coalescing's second part as the object will always contain at least `mintNft: true`

**Review by attemka — APPROVED** [review]

**Pull request merged**